src/app/client/src/app/modules/observation/components/observation-details/observation-details.component.spec.ts

```diff
@@ -1,25 +1,309 @@
+      schemas: [NO_ERRORS_SCHEMA]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(ObservationDetailsComponent);
     component = fixture.componentInstance;
+    activatedRouteStub = TestBed.get(ActivatedRoute);
+    observationUtilService = TestBed.get(ObservationUtilService);
+    observationService = TestBed.get(ObservationService);
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+
+  it('should create', () => {
     expect(component).toBeTruthy();
   });
-});
+
+  it("should fetch profile details on page load", () => {
+    spyOn(observationUtilService, "getProfileDataList").and.callFake(() => Promise.resolve(ProfileData));
+    payload = ProfileData;
+    component.getProfileData();
+    spyOn(component, "getProfileData").and.callThrough();
+    expect(observationUtilService).toBeDefined();
+    expect(observationUtilService).toBeTruthy();
+
+  });
+
+  it('Should fetch Entity list', () => {
+    spyOn(observationService, 'post').and.callFake(() => observableOf(EntityList));
+    spyOn(component, 'getEntities').and.callThrough();
+    spyOn(component, "getObservationForm");
+    component.getEntities();
+    expect(component.getEntities).toHaveBeenCalled();
+    expect(component.selectedEntity).toBe(EntityList.result.entities[0]);
+    expect(component.entities).toBeDefined();
+    expect(component.entities.entities.length).toBeGreaterThan(1);
+    expect(component.getObservationForm).toHaveBeenCalled();
+  });
+
+
+
+  it('Should fetch entity throw error', () => {
+    spyOn(observationService, 'post').and.returnValue(observableThrowError({}));
+    spyOn(component, 'getEntities').and.callThrough();
+    component.getEntities();
+  })
+
+
+  it('Should fetch Observation forms', () => {
+    spyOn(observationService, 'post').and.callFake(() => observableOf(ObservationForm));
+    spyOn(component, 'getObservationForm');
+    component.getObservationForm();
+    component.showLoader = false;
+    component.submissions = ObservationForm.result;
+    expect(component.submissions).toBeDefined();
+    expect(component.submissions.length).toBeGreaterThanOrEqual(0);
+    expect(component.getObservationForm).toHaveBeenCalled();
+  })
+
+
+  it('Should Delete Entity', async () => {
+    spyOn(observationUtilService, "getAlertMetaData").and.callFake(() => AlertMetaData);
+    spyOn(observationUtilService, "showPopupAlert").and.callFake(() => Promise.resolve(true))
+    spyOn(observationService, 'delete').and.callFake(() => of(true));
+    spyOn(component, "delete").and.callThrough();
+    spyOn(observationUtilService, "getProfileDataList").and.callFake(() => Promise.resolve(ProfileData));
+    component.payload = ProfileData;
+    component.payload.data = Entity.data._id;
+    await component.delete(Entity.data);
+    // component.getEntities();
+    setTimeout(() => {
+      expect(observationUtilService).toBeDefined();
```

Comment:
would be better if expect some method of observationUtilService is expected to have been called , instead of toBeDefined